### ISSUE TITLE
Unable to download to virtual fs using pyfetch

### ISSUE THREAD

**Issue by wx4stg** [pyodide/pyodide#2845]
## 🐛 Bug

When downloading files, pyfetch returns a response with a status code of 0. I'm attempting to follow the documentation at https://pyodide.org/en/stable/usage/faq.html#how-can-i-load-external-files-in-pyodide

### To Reproduce

```
<py-script>
    from pyodide.http import pyfetch
    import asyncio
    response = await pyfetch("https://pyodide.org/en/stable/_static/pyodide-logo.png", mode="no-cors")
    console.log(response.status)
</py-script>
```
This will never give "200"

### Expected behavior

The file response to give code 200 and let me save the file

### Environment

- Pyodide Version: 0.20.0
- Browser version: Brave Version 1.40.113 Chromium: 103.0.5060.114 (Official Build) (x86_64)

**Comment by rth** [comment]
Thanks for the report. So far I can't reproduce. Running,
```
>>> from pyodide.http import pyfetch
>>> response = await pyfetch("https://pyodide.org/en/stable/_static/pyodide-logo.png", mode="no-cors")
>>> response.status
200
```
in the [pyodide REPL](https://pyodide.org/en/stable/console.html) v0.20.0 does work as expected. Could you try running there? Otherwise maybe something related to console.log though that also sounds unlikely.

**Comment by rth** [comment]
Though actually yes, locally, one would also get this error state with the following example
<details>

```
<!DOCTYPE html>
<html>
  <head>
      <script src="https://cdn.jsdelivr.net/pyodide/v0.20.0/full/pyodide.js"></script>
  </head>
  <body>
    <script type="text/javascript">
      let pyodide; 
                         
      async function main(){
        pyodide = await loadPyodide();    
        console.log(await pyodide.runPythonAsync(` 
    from pyodide.http import pyfetch                                     
    import asyncio                 
    response = await pyfetch("https://pyodide.org/en/stable/_static/pyodide-logo.png", mode="no-cors")                         
        `));
        console.log(pyodide.runPython("response.status"));
      }
      main();
    </script>
  </body>
</html>
```

</details>

So there are two problems,
 1. fetching fails due to CORS. For instance, doing it directly in JS from a different origin
```
let res = await fetch("https://pyodide.org/en/stable/_static/pyodide-logo.png");
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://pyodide.org/en/stable/_static/pyodide-logo.png. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 200.

Uncaught (in promise) TypeError: NetworkError when attempting to fetch resource. 
```
this is expected, as that file isn't allowed ot be loaded from a different origin than pyodide.org due to CORS headers.

2. That we actually never see this CORS error, and just get the 0 error status:  the `response.status_text` is empty, `await reponse.string()` is empty etc. So we have an issue with error reporting here.

**Comment by wx4stg** [comment]
ah, I see. I'm very new to this web development stuff and was trying to use pyscript to try to avoid learning javascript. I didn't (and still don't) know much about CORS and Same Origin Policy, etc. It seems I might be better off just learning...

Since there is an error reporting issue (problem 2, as you mentioned), I'll not close this

**Comment by danielkauffmann** [comment]
Any update?

**Comment by danielkauffmann** [comment]
Any update?

Reproduced in [pyodide REPL](https://pyodide.org/en/stable/console.html):

![image](https://user-images.githubusercontent.com/62577994/225443736-1ad87223-4ea1-4d61-9247-808a5591ad19.png)

**Comment by ryanking13** [comment]
@danielkauffmann Well, I think that is expected behavior. Since `https://github.com` blocks cross-origin requests. If you use JS `fetch` in the browser with same parameters, you'll get the same result.

Note that ["no-cors" is not a magical keyword](https://stackoverflow.com/questions/43262121/trying-to-use-fetch-and-pass-in-mode-no-cors) that helps you to bypass CORS policy.

If you have to send a cross-origin request you need to 

1. ask the site owner to allow cross-origin requests or 
2. use a proxy server like [cors.sh](https://cors.sh/)

Closing as answered, but don't hesitate to comment further.

**Comment by hoodmane** [comment]
Before this was left open because pyfetch should probably throw an error or be a bit more noisy about the coors issue.

**Comment by rth** [comment]
Yeah, the bug was more about error reporting as far as I understand.
